src/app/users/associateRoles.js

```diff
@@ -92,6 +97,15 @@ const post = async (req, res) => {
   }
 };
 
+const haveRolesBeenUpdated= (req, currentService, selectedRoles) => {
+  if(req.session.user.services
+    && req.session.user.services[currentService].roles
+    && req.session.user.services[currentService].roles.length === selectedRoles.length){
```

Comment:
You are simply checking the number of roles selected matches the current number of roles, this means I could change the role I have selected to a different one but I wouldn't see the review changes page if the length was the same.

You need to check for the items in the array selectedRoles to see if they match what is in the req.session.user.services[currentService].roles array

Reply:
Now doing indepth comparison of the session roles and selected roles